scripts/test-all

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+# This script runs all the tests under a variety of conditions.
+# This should pass before submitting pull requests.
+# Arguments given will be passed through to "cargo test"
+# This runs in the Makefile environment via "make run"
+
+set -euo pipefail
+
+# Run from the Makefile environment
+MAKEFILE_RUN=${MAKEFILE_RUN:-""}
```

Comment:
This seems unnecessary.

Reply:
This is necessary due to `set -u`. Shell scripts should use `set -u` for safety. This is the approach that I use to declare variables before testing them.